packages/typography/src/Link/Link.tsx

```diff
@@ -65,19 +65,30 @@ const Link = InferredPolymorphic<LinkProps, 'span'>(
       }
     }, [rest, currentHostname]);
 
-    let target, rel, icon;
+    let icon;
+
+    const defaultAnchorProps: Pick<
+      JSX.IntrinsicElements['a'],
+      'target' | 'rel'
+    > = {
+      target: undefined,
+      rel: undefined,
+    };
 
-    // Sets defaults for target and rel props when Component is an anchor tag
-    if (Component === 'a') {
+    if ((rest as AnchorLikeProps).target || (rest as AnchorLikeProps).rel) {
```

Comment:
This check is getting a bit verbose. Could consider extracting it to it's own variable
(i.e. `const hasTargetOrRel = (rest as AnchorLikeProps).target || (rest as AnchorLikeProps).rel)`)